gitclone/checkout_helper.go

```diff
@@ -138,7 +139,8 @@ func mergeWithCustomRetry(gitCmd git.Git, arg string, retry fallbackRetry) error
 			return runner.Run(gitCmd.Merge(arg))
 		}
 
-		return fmt.Errorf("merge failed (%s): %v", arg, mErr)
+		wErr := fmt.Errorf("merge failed (%s): %w", arg, mErr)
+		return fmt.Errorf("%v%w", wErr, errors.New("please try to resolve all conflicts between the base and compare branches"))
```

Comment:
NOTE: Currently the final error works well when printed using `errorutil.FormattedError`.

But if you don't use the FormattedError, then the error looks strange a bit:

```
arg := "hash"
mErr := errors.New("couldn't merge")
->
merge failed (hash): couldn't mergeplease try to resolve all conflicts between the base and compare branches
```

If the final error is constructed like this: `return fmt.Errorf("%v: %w", wErr, errors.New("please try to resolve all conflicts between the base and compare branches"))`

it works also well with the `errorutil.FormattedError` and also when simply printed:

`merge failed (hash): couldn't merge: please try to resolve all conflicts between the base and compare branches`

Reply:
[c6249f9](https://github.com/bitrise-steplib/steps-git-clone/pull/210/commits/c6249f93873f4dab79baed792c1d26707c1d2584)